Sources/NIOExtras/LengthFieldBitLength.swift

```diff
@@ -0,0 +1,69 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2021 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+/// An struct to describe the length of a piece of data in bits
+public struct LengthFieldBitLength {
```

Comment:
very annoyingly, we need to prefix this with `NIO*` or move it into one of the pre-existing types.

Reply:
Changed, but why? `LengthFieldBitLength` is not an enum, but a struct.
How does this effect API stability?

Reply:
@dnadoba We [guarantee](https://github.com/apple/swift-nio/blob/main/docs/public-api.md#1-no-global-namespace-additions-that-arent-prefixed) that we won't add _anything_ that doesn't start with `NIO`/`CNIO`/... The reason for that is if a user writes this program:

```
import NIO
import SomethingElse

let l = LengthFieldBitField()
```

and let's assume that `SomethingElse` also provides a `LengthFieldBitField`, then the code would compile _until_ your PR gets merged and then suddenly `LengthFieldBitLength` becomes ambiguous. You, you can change it to `SomethingElse.LengthFieldBitLength` but that requires a code change and the compilation failed.

Therefore we guarantee to only add `NIO*` stuff and then this can't happen assuming that nobody else produces `NIO*` names.

Reply:
What would obviously be awesome is if SwiftPM & Swift would only actually export the `NIO` API given the restrictions in `Package.swift`. If that were the case, then if you wrote `from: "2.0.0"` you would only get the NIO API introduced in 2.0.0 and not what came in 2.1.0 and later even if (say) you compiled against NIO 2.25.0. That would also solve the issue that people just put `from: "2.0.0"` but then end up using much newer NIO features.

But alas this feature isn't actually implemented so we have to use the `NIO*` prefixing hack and hope for the best :).

Reply:
Thank you for your detailed explanation :)